src/App.php

```diff
@@ -1143,7 +1143,7 @@ private function outputResponseJSON($data, array $headers = []): void
     public function getRenderedModals()
     {
         $modals = [];
-        foreach ($this->html->elements as $view) {
+        foreach ($this->html->elements ?? [] as $view) {
```

Comment:
This is not good if `$this->html` may be null. If so use something like `$this->html !== null ? $this->html->elements : []`.

Reply:
while traversing it get [] on every check isset

Reply:
Yes, but elements should not be empty, if so, it should be not supressed.

Reply:
Ok, is not the right way to use it but if i use App this way :
```php
$app = new App();
$app->terminateJSON(['code' => 201, 'message' => 'record created']);
```

i don't need layout and i don't init, just emit a response like a JSON API.
Just now i see that probably this approach can be better
```php
$app = new App();
$app->outputResponseJSON(['code' => 201, 'message' => 'record created']);
$app->callExit();
```
What do you think?

Reply:
I am neutral on that, it has + and -, but terminateXX is handy to enforce it is not called multiple times (which makes sense for JSON - and for custom output streamer, you can still use outputXX()...). What are you solving exactly?